errdefs/defs.go

```diff
@@ -51,6 +51,7 @@ type ErrNotImplemented interface {
 // ErrUnknown signals that the kind of error that occurred is not known.
 type ErrUnknown interface {
 	Unknown()
+	Unwrap() error
```

Comment:
My concern here would be that currently, the types are detected only by the `Unknown()` function being present (to match the interface). Adding `Unwrap()` to the interface could mean there's errors generated that are no nonger satisfying the interface, and thus longer detected as this error-type. (I recall having a discussion at some point to add `Error()` to the interface, so that it would only match if it implemented the `error` interface)

Does it actually have to be added to the _interface_, or only to the `errdefs/helpers` ?

/cc @cpuguy83

Reply:
We do not need this define don the interface types.

If we do add `Unwrap()`, we should add it to all the implementations (but not the interfaces).

Reply:
yes indeed, the `Unwrap()` is not needed in the interface